boxsdk/object/base_object.py

```diff
@@ -49,15 +49,17 @@ def __init__(self, session, object_id, response_object=None):
         super(BaseObject, self).__init__(session)
         self._object_id = object_id
         self._response_object = response_object or {}
-
-    def __getattr__(self, item):
-        """Base class override. Try to get the attribute from the API response object."""
-        return self._response_object[item]
+        self.__dict__.update(self._response_object)
 
     def __getitem__(self, item):
         """Base class override. Try to get the attribute from the API response object."""
         return self._response_object[item]
 
+    def __repr__(self):
+        """Base class override. Return a human-readable representation using the Box ID or name of the object."""
+        description = self.name if 'name' in self._response_object else self.object_id  # pylint:disable=no-member
```

Comment:
We should always show the Box ID. And then also add the `name` if it is available.

We should also always show the `type`. ``self.__class__.__name__`` isn't always descriptive enough, if we haven't written a class / translator for that type.